core/src/main/java/org/mule/api/security/tls/RestrictedSSLSocketFactory.java

```diff
@@ -100,4 +102,24 @@ private SSLSocket restrictCipherSuites(SSLSocket socket)
         socket.setEnabledProtocols(enabledProtocols);
         return socket;
     }
+
+    public static synchronized SocketFactory getDefault()
+    {
+        if(defaultSocketFactory != null)
```

Comment:
I think it would make more sense to check for null and in that case create and assign the field. Then you'd only have one return.